src/tests/ilverify/ILTests/ExceptionRegionTests.il

```diff
@@ -732,5 +732,59 @@
     MethodEnd:
         ret
     }
-}
 
+    .method public hidebysig instance void Catch.ExceptionDerivedType_Valid() cil managed
+    {
+        .maxstack 1
+
+        .try {}
+        catch [System.Runtime]System.InvalidOperationException
+        {
+            leave.s MethodEnd
+        }
+
+        MethodEnd:
+            ret
+    }
+
+    .method public hidebysig instance void Throw.NonExceptionDerivedType_Valid() cil managed
+    {
+        .maxstack 1
+
+        newobj instance void [System.Runtime]System.InvalidOperationException::.ctor()
+        throw
+        ret
+    }
+
+    .method public hidebysig instance void Throw.Null_Valid() cil managed
+    {
+        .maxstack 1
+
+        ldnull
```

Comment:
This is something you can write in C# just fine. 

I think the rule for these sanity checks should be that they never trigger on anything produced by C# compiler.

Reply:
Valid test cases can be written in C#. Do we still need invalid cases written in IL?

Reply:
Yes. We do not want to depend on C# compiler producing a specific IL shape that we want to test.

Reply:
I am unable to run ilverify test project locally #43967, it does not seem to be running in CI either. Under current state, changes in `ILVerification.Tests.csproj` and friends are risky. I can follow up on moving the two valid cases to C# project, once #43967 is resolved.

Reply:
>  it does not seem to be running in CI either.

They do not run in CI by default because they are marked `<CLRTestPriority>1</CLRTestPriority>`. The Priority 1 test run can be triggered manually; or you can make the tests run by flipping the priority to 0 temporarily.

>  can follow up on moving the two valid cases to C# project

I do not think it is something we want to do per my above comment.

Reply:
Please take a look at #43967. This test project is not building properly in local machine, since the last fix.

Reply:
Added a C# project and moved three valid cases there. Only two invalid cases are added in ilproj.